swarm/network/simulation/kademlia.go

```diff
@@ -66,10 +67,10 @@ func (s *Simulation) WaitTillHealthy(ctx context.Context, kadMinProxSize int) (i
 				h := k.Healthy(pp)
 				//print info
 				log.Debug(k.String())
-				log.Debug("kademlia", "empty bins", pp.EmptyBins, "gotNN", h.GotNN, "knowNN", h.KnowNN, "full", h.Full)
-				log.Debug("kademlia", "health", h.GotNN && h.KnowNN && h.Full, "addr", hex.EncodeToString(k.BaseAddr()), "node", id)
-				log.Debug("kademlia", "ill condition", !h.GotNN || !h.Full, "addr", hex.EncodeToString(k.BaseAddr()), "node", id)
-				if !h.GotNN || !h.Full {
+				log.Debug("kademlia", "connectNN", h.ConnectNN, "knowNN", h.KnowNN)
+				log.Debug("kademlia", "health", h.ConnectNN && h.KnowNN, "addr", hex.EncodeToString(k.BaseAddr()), "node", id)
+				log.Debug("kademlia", "ill condition", !h.ConnectNN, "addr", hex.EncodeToString(k.BaseAddr()), "node", id)
+				if !h.ConnectNN {
```

Comment:
Why don't we need to check all the `Saturation` and `Potent` etc. members here? .E.g
`h.ConnectNN && h.KnowNN && h.CountKnowNN > 0`

Reply:
Yes, it should probably have `Potent.` Actually, we should probably discuss with @janos exactly what and how we wait for.

Saturation has to do with _discovery_ of peers, and does not affect "healthy" in the way we use the term.